### PR TITLE
Remove special routing handling for multiple consuming segments

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
- Remove RealtimeSegmentSelector because:
  - HLC is completely deprecated, no need to handle it
  - When there are 2 consuming segments, we want to query both of them for the following reasons:
    - The hotspot issue is already resolved with #10466
    - Comparing to the damage of having overlapping data (which is super unlikely if not impossible), skipping the latest consuming segment when one server stuck can cause missing data for long time, which is much worse
    - Skipping the latest consuming segment can cause wrong result for upsert table
- Add more logs in routing manager for important information such as unavailable instance, new segments, etc.

### CODE REVIEW

File: pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java [hunk 1]
```diff
@@ -314,8 +316,16 @@ void refreshSegmentStates() {
       }
       if (!enabledCandidates.isEmpty()) {
         instanceCandidatesMap.put(segment, enabledCandidates);
+      } else {
+        // Do not count new segment as unavailable
```
Comment (snleee) [inline pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java:320]: `count new segment as available` would be easier to understand

File: pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java [hunk 2]
```diff
@@ -314,8 +316,16 @@ void refreshSegmentStates() {
...
+        List<String> candidateInstances = new ArrayList<>(candidates.size());
+        for (SegmentInstanceCandidate candidate : candidates) {
+          candidateInstances.add(candidate.getInstance());
+        }
+        LOGGER.info("Failed to find servers hosting new segment: {} for table: {} "
+                + "(all candidate instances: {} are disabled, not counting segment as unavailable)", segment,
```
Comment (snleee) [inline pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java:326]: same here.
`all candidate instances: {} are disabled, but counting segment as available for new segment`

**Review by snleee — APPROVED** [review]
- Does the `DefaultSegmentSelector` cover the case that we faced in the production? If not, let's add the testing to cover the scenario.
- Thank you for cleaning up the `RealtimeSegmentSelector` 👍

File: pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java:320]: We don't track available segment but only unavailable ones, so IMO it is more clear to comment on whether it is counted as unavailable

File: pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java:326]: Same as above, but revised the log to be easier to differentiate with old segment one

**Comment by Jackie-Jiang** [comment]
@snleee After removing the special handling, `DefaultSegmentSelector` becomes a simple setter and getter for one field and there is no meaningful way to add a test

**Pull request merged**